pkg/storaged/fsys-tab.jsx

```diff
@@ -95,18 +95,54 @@
     return blocks;
 }
 
+function find_children_for_mount_point(client, mount_point, self) {
+    const children = {};
+
+    function is_self(b) {
+        return self && (b == self || client.blocks[b.CryptoBackingDevice] == self);
+    }
+
+    for (const p in client.blocks) {
+        const b = client.blocks[p];
+        const fs = client.blocks_fsys[p];
+
+        if (is_self(b))
+            continue;
+
+        if (fs) {
+            for (const mp of fs.MountPoints) {
+                const mpd = utils.decode_filename(mp);
+                if (mpd.length > mount_point.length && mpd.indexOf(mount_point) == 0 && mpd[mount_point.length] == "/")
+                    children[mpd] = b;
+            }
+        }
+    }
+
+    return children;
+}
+
 function nice_block_name(block) {
     return utils.block_name(client.blocks[block.CryptoBackingDevice] || block);
 }
 
-export function is_valid_mount_point(client, block, val) {
+export async function is_valid_mount_point(client, block, val) {
     if (val === "")
         return _("Mount point cannot be empty");
 
     const other_blocks = find_blocks_for_mount_point(client, val, block);
     if (other_blocks.length > 0)
         return cockpit.format(_("Mount point is already used for $0"),
                               other_blocks.map(nice_block_name).join(", "));
+
+    const children = find_children_for_mount_point(client, val, block);
+    if (Object.keys(children).length > 0)
+        return <>
+            {_("Filesystems are already mounted below this mountpoint.")}
+            <ul />
+            {Object.keys(children).map(m => <li key={m}>{nice_block_name(children[m])} on {m}</li>)}
+            <ul />
+            {_("Please unmount them first.")}
```

Comment:
These 6 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19388-20230925-134144-be2f6638-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)